src/worker/rpc.rs

```diff
@@ -169,6 +172,28 @@ pub async fn run_worker(
         }
     };
 
+    let hw_state_updater = async move {
```

Comment:
Let's extract this to a separate `async` function that takes `hw_state_poll_interval` and state ref.

Reply:
Done!